test/jasmine/tests/animate_test.js

```diff
@@ -591,32 +591,84 @@ describe('Test animate API', function() {
 describe('Animate API details', function() {
     'use strict';
 
-    var gd, mockCopy;
+    var gd;
+    var dur = 30;
+    var mockCopy;
 
     beforeEach(function(done) {
         gd = createGraphDiv();
-
         mockCopy = Lib.extendDeep({}, mock);
-
-        Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(function() {
-            return Plotly.addFrames(gd, mockCopy.frames);
-        }).then(done);
+        Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
     });
 
     afterEach(function() {
         Plotly.purge(gd);
         destroyGraphDiv();
     });
 
-    it('null frames should not break everything', function(done) {
-        gd._transitionData._frames.push(null);
+    it('redraws after a layout animation', function(done) {
+        var redraws = 0;
+        gd.on('plotly_redraw', function() {redraws++;});
 
-        Plotly.animate(gd, null, {
-            frame: {duration: 0},
-            transition: {duration: 0}
+        Plotly.animate(gd,
+            {layout: {'xaxis.range': [0, 1]}},
+            {frame: {redraw: true, duration: dur}, transition: {duration: dur}}
+        ).then(function() {
+            expect(redraws).toBe(1);
```

Comment:
Not quite, but similar. Sometimes it was zero because the promise resolved before *both* the frame and the transition were registered as complete. Now the promise doesn't resolve until both are done—which shouldn't be more than a couple ms difference, but that's enough to mess up the ordering.